src/app.rs

```diff
@@ -274,7 +279,14 @@ impl App {
             return Err(ServerError::UnreducedCommitment);
         }
 
-        let (tx, rx) = oneshot::channel();
+        let identity_exists = self.database.identity_exists(commitment).await?;
+        if identity_exists {
+            return Err(ServerError::DuplicateCommitment);
+        }
+
+        self.database.insert_new_identity(commitment).await?;
+
+        let (tx, _) = oneshot::channel();
```

Comment:
We should get rid of this channel entirely and not send anything from the insert_identities task